modules/appeals_api/app/controllers/appeals_api/oauth_application_controller.rb

```diff
@@ -0,0 +1,175 @@
+# frozen_string_literal: true
+
+module AppealsApi
+  class OAuthApplicationController < ::OpenidApplicationController
+    include AppealsApi::MPIVeteran
+    before_action :verify_oauth_token!
+    before_action :verify_oauth_scopes!
+
+    # HEADERS should be a list of header names of interest, defined in the parent
+    # controller based on the form schema.
+    #
+    # @return [Hash<String,String>] request headers of interest as a hash
+    def request_headers
+      return {} unless defined? self.class::HEADERS
+
+      self.class::HEADERS.index_with { |key| request.headers[key] }.compact
+    end
+
+    #
+    # Override this in individual controllers if needed to return a list of required OAuth
+    # scopes based on the context.
+    # FIXME: replace these with actual scopes
+    #
+    # @return [Array<String>] OAuth scopes required for a successful current request
+    def oauth_scopes
+      case action_name
+      when 'index'
+        %w[claim.read]
+      when 'show'
+        %w[claim.read]
+      when 'create'
+        %w[claim.write]
+      else
+        []
+      end
+    end
+
```

Comment:
Most of this is very closely based on this code in [this controller](https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/claims_api/app/controllers/claims_api/v2/application_controller.rb) and [this concern](https://github.com/department-of-veterans-affairs/vets-api/blob/be114fe97c1d6e7d87b653ed77bc4f2ae75a7828/modules/claims_api/app/controllers/concerns/claims_api/ccg_token_validation.rb) from the Claims API